include/boost/unordered/unordered_flat_map.hpp

```diff
@@ -61,11 +66,26 @@ namespace boost {
         }
 
         template <class A, class... Args>
-        static void construct(A& al, element_type* p, Args&&... args)
+        static void construct(A& al, init_type* p, Args&&... args)
         {
           boost::allocator_construct(al, p, std::forward<Args>(args)...);
         }
 
+        template <class A, class... Args>
+        static void construct(A& al, element_type* p, Args&&... args)
+        {
```

Comment:
This looks inconsistent: why are you rebinding here and not in `construct(A& al, init_type* p, Args&&... args)`? I understand the contract is that the caller passes an appropriately rebound allocator.

Reply:
Also, consider renaming `construct(A& al, element_type* p, Args&&... args)` and `destroy(A& al, element_type* p)` to `construct(A& al, value_type* p, Args&&... args)` and `destroy(A& al, value_type* p)`: for flat containers, `element_type` is a mere alias to `value_type`  so it looks nicer if `element_type` does not appear in the policy except for its definition (unlike in node containers, where `element_type` receives special treatment).

Reply:
Good catch. I was erroneously rebinding, likely as an artifact of early passes at this refactor.

The code has been updated such that it no longer needs to rebind at all. We know that our Allocator is properly bound to allocate, deallocate value_type. The construct(), destroy() methods are supposed to work with any and all types without the need for rebinding.